### ISSUE TITLE
Failed to rebuild org.apache.jspwiki:jspwiki-builder:2.11.3

### ISSUE THREAD

**Issue by tromai** [jvm-repo-rebuild/reproducible-central#662]
Hi, 

Thank you for your great work on `reproducible-central`. We have been trying to run the rebuild using Reproducible Central (RC) 's infrastructure on this build spec https://github.com/jvm-repo-rebuild/reproducible-central/blob/master/content/org/apache/jspwiki/jspwiki-2.11.3.buildspec but failed. We would love to have your insights on what could be the issue and how we can resolve it. 

# Reproducing the error
## System information
- Oracle Linux 9.4  running in WSL2
- Podman version 4.9.4-rhel
- Reproducible central at commit 29436ffbd03f319179127e3a5f9984f796b32fd8

## Commands
```bash
$ git checkout https://github.com/jvm-repo-rebuild/reproducible-central.git
$ cd reproducible-central
$ git checkout 29436ffbd03f319179127e3a5f9984f796b32fd8
$ export BUILDAH_FORMAT=docker
$ RB_OCI_ENGINER="podman" RB_OCI_ENGINE_RUN_OPTS="--userns=keep-id" RB_OCI_VOLUME_FLAGS=":Z,rw" ./rebuild.sh content/org/apache/jspwiki/jspwiki-2.11.3.buildspec
```

## Behavior
The toolchain Docker image was created successfully, but the maven build command failed with the following error

<details>

<summary>Errors and stacktraces</summary>

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  29.445 s
[INFO] Finished at: 2024-11-14T23:44:54Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal ro.isdc.wro4j:wro4j-maven-plugin:1.8.0:run (buildPlainEditorSyntaxJS) on project jspwiki-main: Execution buildPlainEditorSyntaxJS of goal ro.isdc.wro4j:wro4j-maven-plugin:1.8.0:run failed: Plugin ro.isdc.wro4j:wro4j-maven-plugin:1.8.0 or one of its dependencies could not be resolved: Could not find artifact me.n4u.sass:sass-gems:jar:3.1.19 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal ro.isdc.wro4j:wro4j-maven-plugin:1.8.0:run (buildPlainEditorSyntaxJS) on project jspwiki-main: Execution buildPlainEditorSyntaxJS of goal ro.isdc.wro4j:wro4j-maven-plugin:1.8.0:run failed: Plugin ro.isdc.wro4j:wro4j-maven-plugin:1.8.0 or one of its dependencies could not be resolved: Could not find artifact me.n4u.sass:sass-gems:jar:3.1.19
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution buildPlainEditorSyntaxJS of goal ro.isdc.wro4j:wro4j-maven-plugin:1.8.0:run failed: Plugin ro.isdc.wro4j:wro4j-maven-plugin:1.8.0 or one of its dependencies could not be resolved: Could not find artifact me.n4u.sass:sass-gems:jar:3.1.19
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:109)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.PluginResolutionException: Plugin ro.isdc.wro4j:wro4j-maven-plugin:1.8.0 or one of its dependencies could not be resolved: Could not find artifact me.n4u.sass:sass-gems:jar:3.1.19
    at org.apache.maven.plugin.internal.DefaultPluginDependenciesResolver.resolveInternal (DefaultPluginDependenciesResolver.java:221)
    at org.apache.maven.plugin.internal.DefaultPluginDependenciesResolver.resolve (DefaultPluginDependenciesResolver.java:152)
    at org.apache.maven.plugin.internal.DefaultMavenPluginManager.createPluginRealm (DefaultMavenPluginManager.java:404)
    at org.apache.maven.plugin.internal.DefaultMavenPluginManager.setupPluginRealm (DefaultMavenPluginManager.java:374)
    at org.apache.maven.plugin.DefaultBuildPluginManager.getPluginRealm (DefaultBuildPluginManager.java:234)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:105)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not find artifact me.n4u.sass:sass-gems:jar:3.1.19
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:424)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:229)
    at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies (DefaultRepositorySystem.java:340)
    at org.apache.maven.plugin.internal.DefaultPluginDependenciesResolver.resolveInternal (DefaultPluginDependenciesResolver.java:213)
    at org.apache.maven.plugin.internal.DefaultPluginDependenciesResolver.resolve (DefaultPluginDependenciesResolver.java:152)
    at org.apache.maven.plugin.internal.DefaultMavenPluginManager.createPluginRealm (DefaultMavenPluginManager.java:404)
    at org.apache.maven.plugin.internal.DefaultMavenPluginManager.setupPluginRealm (DefaultMavenPluginManager.java:374)
    at org.apache.maven.plugin.DefaultBuildPluginManager.getPluginRealm (DefaultBuildPluginManager.java:234)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:105)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact me.n4u.sass:sass-gems:jar:3.1.19
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:414)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:229)
    at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies (DefaultRepositorySystem.java:340)
    at org.apache.maven.plugin.internal.DefaultPluginDependenciesResolver.resolveInternal (DefaultPluginDependenciesResolver.java:213)
    at org.apache.maven.plugin.internal.DefaultPluginDependenciesResolver.resolve (DefaultPluginDependenciesResolver.java:152)
    at org.apache.maven.plugin.internal.DefaultMavenPluginManager.createPluginRealm (DefaultMavenPluginManager.java:404)
    at org.apache.maven.plugin.internal.DefaultMavenPluginManager.setupPluginRealm (DefaultMavenPluginManager.java:374)
    at org.apache.maven.plugin.DefaultBuildPluginManager.getPluginRealm (DefaultBuildPluginManager.java:234)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:105)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :jspwiki-main

[FAIL] /========================================================================
[FAIL] |  failed to build
[FAIL] \========================================================================
[FAIL]  ---------->>> PROCESS WAS ABORTED <<<----------
```

</details>

# What we tried to resolve the issue
To resolve this issue, I removed all of the mirrors configuration in [`reproducible-central/m2/settings.xml`](https://github.com/jvm-repo-rebuild/reproducible-central/blob/master/m2/settings.xml#L86-L256) and then run the exact rebuild command again
```bash
RB_OCI_ENGINER="podman" RB_OCI_ENGINE_RUN_OPTS="--userns=keep-id" RB_OCI_VOLUME_FLAGS=":Z,rw" ./rebuild.sh content/org/apache/jspwiki/jspwiki-2.11.3.buildspec
```
This time, the build passed.

<details>

<summary>Build log</summary>

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:09 min
[INFO] Finished at: 2024-11-14T23:53:12Z
[INFO] ------------------------------------------------------------------------
dos2unix: converting file target/jspwiki-builder-2.11.3.buildinfo to Unix format...
dos2unix: converting file target/jspwiki-builder-2.11.3.buildcompare to Unix format...

[INFO] rebuild from content/org/apache/jspwiki/jspwiki-2.11.3.buildspec
[INFO]   results in content/org/apache/jspwiki/jspwiki-builder-2.11.3.buildinfo
[INFO] compared to Central Repository content/org/apache/jspwiki/jspwiki-builder-2.11.3.buildcompare:
           ok=77
           okFiles="jspwiki-builder-2.11.3.pom jspwiki-wikipages-builder-2.11.3.pom jspwiki-wikipages-de-2.11.3.pom jspwiki-wikipages-en-2.11.3.pom jspwiki-wikipages-es-2.11.3.pom jspwiki-wikipages-fi-2.11.3.pom jspwiki-wikipages-fr-2.11.3.pom jspwiki-wikipages-it-2.11.3.pom jspwiki-wikipages-nl-2.11.3.pom jspwiki-wikipages-pt_BR-2.11.3.pom jspwiki-wikipages-ru-2.11.3.pom jspwiki-wikipages-zh_CN-2.11.3.pom jspwiki-util-2.11.3.pom jspwiki-util-2.11.3.jar jspwiki-util-2.11.3-sources.jar jspwiki-event-2.11.3.pom jspwiki-event-2.11.3.jar jspwiki-event-2.11.3-sources.jar jspwiki-api-2.11.3.pom jspwiki-api-2.11.3.jar jspwiki-api-2.11.3-sources.jar jspwiki-bootstrap-2.11.3.pom jspwiki-bootstrap-2.11.3.jar jspwiki-bootstrap-2.11.3-sources.jar jspwiki-cache-2.11.3.pom jspwiki-cache-2.11.3.jar jspwiki-cache-2.11.3-sources.jar jspwiki-http-2.11.3.pom jspwiki-http-2.11.3.jar jspwiki-http-2.11.3-sources.jar jspwiki-wysiwyg-2.11.3.pom jspwiki-wysiwyg-2.11.3.jar jspwiki-wysiwyg-2.11.3-sources.jar jspwiki-main-2.11.3.pom jspwiki-main-2.11.3.jar jspwiki-main-2.11.3-sources.jar jspwiki-markdown-2.11.3.pom jspwiki-markdown-2.11.3.jar jspwiki-markdown-2.11.3-sources.jar jspwiki-tika-searchprovider-2.11.3.pom jspwiki-tika-searchprovider-2.11.3.jar jspwiki-tika-searchprovider-2.11.3-sources.jar jspwiki-kendra-searchprovider-2.11.3.pom jspwiki-kendra-searchprovider-2.11.3.jar jspwiki-kendra-searchprovider-2.11.3-sources.jar jspwiki-210-test-adaptees-2.11.3.pom jspwiki-210-test-adaptees-2.11.3.jar jspwiki-210-test-adaptees-2.11.3-sources.jar jspwiki-210-adapters-2.11.3.pom jspwiki-210-adapters-2.11.3.jar jspwiki-210-adapters-2.11.3-sources.jar jspwiki-war-2.11.3.pom jspwiki-war-2.11.3.war jspwiki-war-2.11.3-sources.jar jspwiki-portable-2.11.3.pom jspwiki-portable-2.11.3.jar jspwiki-portable-2.11.3-sources.jar jspwiki-it-builder-2.11.3.pom jspwiki-selenide-tests-2.11.3.pom jspwiki-selenide-tests-2.11.3.jar jspwiki-selenide-tests-2.11.3-sources.jar jspwiki-it-test-custom-2.11.3.pom jspwiki-it-test-custom-2.11.3.war jspwiki-it-test-custom-2.11.3-sources.jar jspwiki-it-test-custom-absolute-urls-2.11.3.pom jspwiki-it-test-custom-absolute-urls-2.11.3.war jspwiki-it-test-custom-absolute-urls-2.11.3-sources.jar jspwiki-it-test-custom-jdbc-2.11.3.pom jspwiki-it-test-custom-jdbc-2.11.3.war jspwiki-it-test-custom-jdbc-2.11.3-sources.jar jspwiki-it-test-cma-2.11.3.pom jspwiki-it-test-cma-2.11.3.war jspwiki-it-test-cma-2.11.3-sources.jar jspwiki-it-test-cma-jdbc-2.11.3.pom jspwiki-it-test-cma-jdbc-2.11.3.war jspwiki-it-test-cma-jdbc-2.11.3-sources.jar jspwiki-bom-2.11.3.pom"
           ko=33
           koFiles="jspwiki-builder-2.11.3-source-release.zip jspwiki-wikipages-de-2.11.3.jar jspwiki-wikipages-de-2.11.3.zip jspwiki-wikipages-de-2.11.3-sources.jar jspwiki-wikipages-en-2.11.3.jar jspwiki-wikipages-en-2.11.3.zip jspwiki-wikipages-en-2.11.3-sources.jar jspwiki-wikipages-es-2.11.3.jar jspwiki-wikipages-es-2.11.3.zip jspwiki-wikipages-es-2.11.3-sources.jar jspwiki-wikipages-fi-2.11.3.jar jspwiki-wikipages-fi-2.11.3.zip jspwiki-wikipages-fi-2.11.3-sources.jar jspwiki-wikipages-fr-2.11.3.jar jspwiki-wikipages-fr-2.11.3.zip jspwiki-wikipages-fr-2.11.3-sources.jar jspwiki-wikipages-it-2.11.3.jar jspwiki-wikipages-it-2.11.3.zip jspwiki-wikipages-it-2.11.3-sources.jar jspwiki-wikipages-nl-2.11.3.jar jspwiki-wikipages-nl-2.11.3.zip jspwiki-wikipages-nl-2.11.3-sources.jar jspwiki-wikipages-pt_BR-2.11.3.jar jspwiki-wikipages-pt_BR-2.11.3.zip jspwiki-wikipages-pt_BR-2.11.3-sources.jar jspwiki-wikipages-ru-2.11.3.jar jspwiki-wikipages-ru-2.11.3.zip jspwiki-wikipages-ru-2.11.3-sources.jar jspwiki-wikipages-zh_CN-2.11.3.jar jspwiki-wikipages-zh_CN-2.11.3.zip jspwiki-wikipages-zh_CN-2.11.3-sources.jar jspwiki-portable-2.11.3-woas.tar.gz jspwiki-portable-2.11.3-woas.zip"
[INFO]     check .buildspec jdk=8 vs reference java.version=1.8 (from MANIFEST.MF Build-Jdk-Spec)
[INFO]     check .buildspec newline=crlf vs reference os.name=Windows (from pom.properties newline) (newline should be crlf if os.name is Windows, lf instead)
[INFO] build available in content/org/apache/jspwiki/buildcache/jspwiki-builder, where you can execute diffoscope
# diffoscope target/reference/org.apache.jspwiki/jspwiki-builder-2.11.3-source-release.zip target/jspwiki-builder-2.11.3-source-release.zip
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-de-2.11.3.jar jspwiki-wikipages/de/target/jspwiki-wikipages-de-2.11.3.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-de-2.11.3.zip jspwiki-wikipages/de/target/jspwiki-wikipages-de-2.11.3.zip
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-de-2.11.3-sources.jar jspwiki-wikipages/de/target/jspwiki-wikipages-de-2.11.3-sources.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-en-2.11.3.jar jspwiki-wikipages/en/target/jspwiki-wikipages-en-2.11.3.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-en-2.11.3.zip jspwiki-wikipages/en/target/jspwiki-wikipages-en-2.11.3.zip
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-en-2.11.3-sources.jar jspwiki-wikipages/en/target/jspwiki-wikipages-en-2.11.3-sources.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-es-2.11.3.jar jspwiki-wikipages/es/target/jspwiki-wikipages-es-2.11.3.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-es-2.11.3.zip jspwiki-wikipages/es/target/jspwiki-wikipages-es-2.11.3.zip
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-es-2.11.3-sources.jar jspwiki-wikipages/es/target/jspwiki-wikipages-es-2.11.3-sources.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-fi-2.11.3.jar jspwiki-wikipages/fi/target/jspwiki-wikipages-fi-2.11.3.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-fi-2.11.3.zip jspwiki-wikipages/fi/target/jspwiki-wikipages-fi-2.11.3.zip
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-fi-2.11.3-sources.jar jspwiki-wikipages/fi/target/jspwiki-wikipages-fi-2.11.3-sources.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-fr-2.11.3.jar jspwiki-wikipages/fr/target/jspwiki-wikipages-fr-2.11.3.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-fr-2.11.3.zip jspwiki-wikipages/fr/target/jspwiki-wikipages-fr-2.11.3.zip
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-fr-2.11.3-sources.jar jspwiki-wikipages/fr/target/jspwiki-wikipages-fr-2.11.3-sources.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-it-2.11.3.jar jspwiki-wikipages/it/target/jspwiki-wikipages-it-2.11.3.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-it-2.11.3.zip jspwiki-wikipages/it/target/jspwiki-wikipages-it-2.11.3.zip
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-it-2.11.3-sources.jar jspwiki-wikipages/it/target/jspwiki-wikipages-it-2.11.3-sources.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-nl-2.11.3.jar jspwiki-wikipages/nl/target/jspwiki-wikipages-nl-2.11.3.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-nl-2.11.3.zip jspwiki-wikipages/nl/target/jspwiki-wikipages-nl-2.11.3.zip
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-nl-2.11.3-sources.jar jspwiki-wikipages/nl/target/jspwiki-wikipages-nl-2.11.3-sources.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-pt_BR-2.11.3.jar jspwiki-wikipages/pt_BR/target/jspwiki-wikipages-pt_BR-2.11.3.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-pt_BR-2.11.3.zip jspwiki-wikipages/pt_BR/target/jspwiki-wikipages-pt_BR-2.11.3.zip
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-pt_BR-2.11.3-sources.jar jspwiki-wikipages/pt_BR/target/jspwiki-wikipages-pt_BR-2.11.3-sources.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-ru-2.11.3.jar jspwiki-wikipages/ru/target/jspwiki-wikipages-ru-2.11.3.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-ru-2.11.3.zip jspwiki-wikipages/ru/target/jspwiki-wikipages-ru-2.11.3.zip
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-ru-2.11.3-sources.jar jspwiki-wikipages/ru/target/jspwiki-wikipages-ru-2.11.3-sources.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-zh_CN-2.11.3.jar jspwiki-wikipages/zh_CN/target/jspwiki-wikipages-zh_CN-2.11.3.jar
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-zh_CN-2.11.3.zip jspwiki-wikipages/zh_CN/target/jspwiki-wikipages-zh_CN-2.11.3.zip
# diffoscope target/reference/org.apache.jspwiki.wikipages/jspwiki-wikipages-zh_CN-2.11.3-sources.jar jspwiki-wikipages/zh_CN/target/jspwiki-wikipages-zh_CN-2.11.3-sources.jar
# diffoscope target/reference/org.apache.jspwiki/jspwiki-portable-2.11.3-woas.tar.gz jspwiki-portable/target/jspwiki-portable-2.11.3-woas.tar.gz
# diffoscope target/reference/org.apache.jspwiki/jspwiki-portable-2.11.3-woas.zip jspwiki-portable/target/jspwiki-portable-2.11.3-woas.zip
[INFO] To see every differences between current rebuild and reference, run:
[INFO]     ./build_diffoscope.sh content/org/apache/jspwiki/jspwiki-builder-2.11.3.buildcompare buildcache/jspwiki-builder
```

</details>

# Extra question
We are also curious on the reasons behind the mirror configurations in this `settings.xml` file. From my understanding, this file is copied into the Docker container where the build runs (see [here](https://github.com/jvm-repo-rebuild/reproducible-central/blob/10c8c2e4f5fdb5a1a4fb52def059e68c31e3f1db/bin/includes/mvnBuildDocker.sh#L99)), which is then used by Maven inside the container.

**Comment by hboutemy** [comment]
wow, very interesting: I can reproduce the issue on my own machine

as you analyzed, this is related to strict check of accepted remote repositories: in your workaround, you disabled the strict check.
I have another option: use Maven 3.9 = in the buildspec, change `mvn` (defaults to 3.6.3) to `mvn-3.9.9`
Root cause: remote repository availability in the dependency tree was broken in Maven 3.6.3, with some edge case doing wrong results. This issue was hard to even report, but was solved in Maven 3.8 or 3.9, with some fixes in Maven Resolver

I'll update the buildspec in Git, because it clearly does not work as is

on the extra question: while checking reproducible builds, I chose to add one layer of tracking = track which external repositories are used by the projects I rebuild, and not let Maven download anything from anywhere (just because it was a nice opportunity to do that: I chose to be picky :) )
In this precise case, it triggered a bug in Maven 3.6.3 that tried to use wrong repositories (and disabled `central`...)

**Comment by tromai** [comment]
Thanks for the response @hboutemy .
> on the extra question: ...

Ah I see, that makes sense.

> This issue was hard to even report, but was solved in Maven 3.8 or 3.9, with some fixes in Maven Resolver

This is indeed quite interesting. I wonder if this is a known bug for Maven that was later fixed or is it totally brand new. We would really love to know more about this Maven issue and whether could you point us to the relevant fixes in Maven Resolver if you happen to know about it?.

I have tried your fix and it works smoothly. This is much better than my work around by removing settings.xml.

In additions, we have also tried the rebuild on a couple more build specs and received similar issues as this case:
1. `content/org/apache/jspwiki/jspwiki-2.11.2.buildspec`: fixing the maven version works
2. `content/org/apache/maven/doxia/doxia/doxia-1.9.1.buildspec`: fixing the maven version works
3. `content/org/apache/plc4x/plc4x/plc4x-0.7.0.buildspec`: fixing the maven version **doesn't work**. This is because the build expects a dependency that is not available in Maven central.

I am planning to create separated Github issues for them with more details. Would this be okay for you? Or would you prefer to keep everything in one issue? 

Looking forward to hearing back from you.

**Comment by hboutemy** [comment]
>  whether could you point us to the relevant fixes in Maven Resolver if you happen to know about it?

I can't find clear pointer, sorry, it was part of cleanup PRs, not really a precise target tracked: given we have a project that fails with older releases, we could probably take time to find the first Maven release that works, to more precisely point to the why...

> I plan to create separate Github issues for each, with additional details. Would this be okay for you? Or would you prefer to keep everything in one issue?

one PR per fix, please, with only `.buildspec` update: we'll let CI commit output updates

thanks for contributing
